src/app/modules/examples/examples.service.ts

```diff
@@ -0,0 +1,327 @@
+        message: 'Error in retrieving example forms',
+        meta: {
+          action: 'execExamplesQuery',
+        },
+        error,
+      })
+
+      return new DatabaseError()
+    },
+  ).andThen((results) => {
+    const formattedResults = results.map((x) => ({
+      ...x,
+      timeText: formatToRelativeString(x.lastSubmission),
+    }))
+
+    return okAsync({ forms: formattedResults })
+  })
+}
+
+/**
+ * Returns the relevant form information (formId, form title, agency logo and
+ * color theme) of the given form id.
+ * @param formId the id of the form to retrieve information for
+ * @returns ok(form info) if form can be found in the database
+ * @returns err(ResultsNotFoundError) if form cannot be found in the database
+ * @returns err(DatabaseError) if error occurs whilst querying the database
+ */
+const getFormInfo = (
+  formId: string,
+): ResultAsync<FormInfo, DatabaseError | ResultsNotFoundError> => {
+  const formIdInfoPipeline = createFormIdInfoPipeline(formId)
+
+  return ResultAsync.fromPromise(
+    FormModel.aggregate<FormInfo>(formIdInfoPipeline)
+      .read('secondary')
+      .allowDiskUse(true)
+      .exec() as Promise<SingleFormInfoQueryResult>,
+    (error) => {
+      logger.error({
+        message: 'Database error retrieving example form info',
+        meta: {
+          action: 'getFormInfo',
+          formId,
+        },
+        error,
+      })
+
+      return new DatabaseError()
+    },
+  ).andThen((result) => {
+    if (!result) {
+      return errAsync(new ResultsNotFoundError('Invalid aggregation pipeline'))
+    }
+    const [retrievedFormInfo] = result
+    // No form data retrieved inside result (formId likely invalid)
+    if (!retrievedFormInfo) {
+      return errAsync(
+        new ResultsNotFoundError(
+          'Error in retrieving template form - form not found.',
+        ),
+      )
+    }
+
+    return okAsync(retrievedFormInfo)
+  })
+}
+
+/**
+ * Retrieves example forms from either the FormStatisticsTotal aggregated
+ * collection or from the Submissions collection via the given type parameter.
+ * @param type the type of collection to retrieve from
+ * @param query query parameters containing data to filter the retrieved examples forms.
+ * @returns ok(object with list of retrieved example forms along with the total number of results) if `shouldGetTotalNumResults` is of string `"true"`
+ * @returns ok(list of retrieved example forms) if `shouldGetTotalNumResults` is not of string `"true"`
+ * @returns err(DatabaseError) if any errors occurs whilst running the pipeline on the database
+ */
+export const getExampleForms = (type: RetrievalType) => (
+  query: ExamplesQueryParams,
+): ResultAsync<QueryPageResultWithTotal | QueryPageResult, DatabaseError> => {
+  const queryBuilder = getExamplesQueryBuilder({ type, query })
+
+  const { pageNo, shouldGetTotalNumResults } = query
+  const offset = parseInt(pageNo) * PAGE_SIZE || 0
+
+  return shouldGetTotalNumResults === 'true'
```

Comment:
I really like what you've done in this function! Imo it's still quite messy because there's so much branching going on that it's hard to keep track though. Ideally we would put all the branching in one function to make all the possibilities clear. But I'm okay to leave it as-is since we're probably going to revamp the examples page anyway, so there's not much value in cleaning up further.